pkg/osvscanner/vulnerability_result.go

```diff
@@ -11,47 +11,74 @@ import (
 	"github.com/google/osv-scanner/pkg/reporter"
 )
 
-// groupResponseBySource converts raw OSV API response into structured vulnerability information
-// grouped by source location.
-func groupResponseBySource(r reporter.Reporter, query osv.BatchedQuery, resp *osv.HydratedBatchedResponse, callAnalysis bool) models.VulnerabilityResults {
+// assembleResult takes the responses from the OSV API and the deps.dev API
+// and converts this into a VulnerabilityResults. As part is this, it groups
+// vulnerability information by source location.
+func assembleResult(
```

Comment:
This seems to be getting pretty long, can you add a TODO comment here to refactor this function in the future?